File: laythe_vm/fixture/rb_benchmark/instantiation.rb
class Foo
  def initialize
  end
end

start = Time.new

500000.times do |_|
  Foo.new
  Foo.new
  Foo.new
  Foo.new
  Foo.new
  Foo.new
  Foo.new
  Foo.new
  Foo.new
  Foo.new
  Foo.new
  Foo.new
  Foo.new
  Foo.new
  Foo.new
  Foo.new
  Foo.new
  Foo.new
  Foo.new
  Foo.new
  Foo.new
  Foo.new
  Foo.new
  Foo.new
  Foo.new
  Foo.new
  Foo.new
  Foo.new
  Foo.new
  Foo.new
end

puts Time.new - start

File: laythe_vm/fixture/py_benchmark/instantiation.py
import time

class Foo:
  def __init__(self):
    pass

start = time.time()

for _ in range(500000):
  Foo()
  Foo()
  Foo()
  Foo()
  Foo()
  Foo()
  Foo()
  Foo()
  Foo()
  Foo()
  Foo()
  Foo()
  Foo()
  Foo()
  Foo()
  Foo()
  Foo()
  Foo()
  Foo()
  Foo()
  Foo()
  Foo()
  Foo()
  Foo()
  Foo()
  Foo()
  Foo()
  Foo()
  Foo()
  Foo()

print(time.time() - start)

